include/alpaka/core/RuntimeMacros.hpp

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2022  Andrea Bocci, Mehmet Yusufoglu
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+//! ALPAKA_DEVICE_THROW either aborts(terminating the program and creating a core dump) or throws std::runtime_error
+//! depending on the Acc. It is impossible to "catch" aborts; but the std::runtime_error exception can be catched in
+//! the catch block.
+//!
+//! The OpenMP specification mandates that exceptions thrown by some thread must be handled by the same thread.
+//! Therefore std::runtime_error thrown by ALPAKA_DEVICE_THROW aborts the the program for OpenMP backends. If needed
+//! the SIGABRT signal can be catched by signal handler.
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && defined(__CUDA_ARCH__)
+#    define ALPAKA_DEVICE_THROW(MSG)                                                                                  \
+        {                                                                                                             \
+            printf("%s [ALPAKA_DEVICE_THROW: Calling __trap(). Cuda backend is enabled.]\n", (MSG));                  \
+            __trap();                                                                                                 \
+        }
+#elif defined(ALPAKA_ACC_GPU_HIP_ENABLED) && defined(__HIP_DEVICE_COMPILE__)
+#    define ALPAKA_DEVICE_THROW(MSG)                                                                                  \
+        {                                                                                                             \
+            printf("%s [ALPAKA_DEVICE_THROW: Calling abort(). Hip backend is enabled.]\n", (MSG));                    \
+            abort();                                                                                                  \
+        }
+#elif defined(ALPAKA_ACC_SYCL_ENABLED) && defined(__SYCL_DEVICE_ONLY__)
+#    define ALPAKA_DEVICE_THROW(MSG)                                                                                  \
+        {                                                                                                             \
+            printf("%s [ALPAKA_DEVICE_THROW: Calling std::abort(). Sycl backend is enabled.]\n", (MSG));              \
+            std::abort();                                                                                             \
```

Comment:
Are you sure we can call `std::abort()` from inside a SYCL kernel running _e.g._ on a GPU ?

Reply:
We only support one-api and it is tested on CI and HAL. .  I wanted to do assert(false) but assert turned out to be calling std::abort from kernel in sycl ([url](https://github.com/intel/llvm/pull/3461/files/641d07178f240f1b3f4d63e9f1836095f274eeae#diff-27b10bd82fa1537002ee64fb2aa6b289ac9aa6527cc17d51b5ceb850c384d202))